src/app-layer-parser.c

```diff
@@ -957,7 +957,8 @@ void AppLayerParserTransactionsCleanup(Flow *f, const uint8_t pkt_dir)
         }
 
         if (txd && has_tx_detect_flags) {
-            if (!IS_DISRUPTED(ts_disrupt_flags) && f->sgh_toserver != NULL) {
+            if (!IS_DISRUPTED(ts_disrupt_flags) &&
+                    (f->sgh_toserver != NULL || (f->flags & FLOW_SGH_TOSERVER) == 0)) {
```

Comment:
Q: Could you please tell in what case should we have a sgh but not the flags set?

Reply:
I do not know any...
Should we test only the flags ?

Reply:
Oh no, we test if (the flag is set) and the pointer is not null, or if the flag is unset